builtin/providers/aws/resource_aws_security_group.go

```diff
@@ -407,6 +411,23 @@ func resourceAwsSecurityGroupUpdateRules(
 		ns := n.(*schema.Set)
 
 		remove := expandIPPerms(group, os.Difference(ns).List())
+
+		if len(remove) == 0 && len(os.List()) == 0 && ruleset == "egress" {
+			// For new Security groups, a default Egress rule is created. Here we add
+			// the equivelent IPPermission struct for removal, so that only the
+			// supplied egress rules exist in the security group.
+			remove = append(remove, &ec2.IPPermission{
+				FromPort: aws.Long(int64(0)),
+				ToPort:   aws.Long(int64(0)),
+				IPRanges: []*ec2.IPRange{
+					&ec2.IPRange{
+						CIDRIP: aws.String("0.0.0.0/0"),
+					},
+				},
+				IPProtocol: aws.String("-1"),
+			})
```

Comment:
Will this work if a user specifies the same egress rule in their config?

Probably an AccTest would answer one way or another.